### ISSUE TITLE
Export png doesn't show the dom-node

### ISSUE THREAD

**Issue by Keinstah** [mwri/cytoscape-dom-node#3]
Using `cy.png()` doesn't produce an image with dom-node

Actual:

![graph (5)](https://user-images.githubusercontent.com/3806194/145524110-9cdc6cb9-e8fd-46e0-8880-99aad34b44cd.png)

Expected:
![image](https://user-images.githubusercontent.com/3806194/145524320-a5fa6934-511a-4785-910e-25887a8114cc.png)

**Comment by mwri** [comment]
You're right, good shout. The problem here is Cytoscape doesn't know anything about the HTML elements, it renders everything on a HTML Canvas, and cytoscape-dom-node puts the DOM elements on top of that. Tricky to solve. Either the DOM elements have to be rendered on the canvas (sounds difficult) or `cy.png` has to be augmented so it renders the DOM elements afterwards (maybe the chief problem here is a reliable way of making an image from a DOM element, there are definitely some bad ways of doing this, just not sure if there is a reliable compatible way).

**Comment by mwri** [comment]
I had a little look at this, I took the codepen demo and made it into a webpack project, and added a snapshots.html which uses html2canvas. It can certainly render the demo well enough. In this form it doesn't support all the options that the cytoscapes png and jpg functions support of course, and no idea how efficient it is, but it shows it should be possible to do something about this in principal.

**Comment by BenBullock1992** [comment]
I too am having this issue. Has any progress been made or any reliable workaround found?

**Comment by mwri** [comment]
@BenBullock1992 Does the above solution / work around not suit you?

https://github.com/mwri/cytoscape-dom-node/blob/7cd93b5a9f4c053ccb57fd54c05b6eb225d04858/demo/src/index.js#L99
